src/main/java/com/cloudbees/plugins/credentials/impl/CertificateCredentialsImpl.java

```diff
@@ -640,7 +648,30 @@ public FormValidation doCheckUploadedKeystore(@QueryParameter String value,
                 if (StringUtils.isBlank(value)) {
                     return FormValidation.error(Messages.CertificateCredentialsImpl_NoCertificateUploaded());
                 }
-                return validateCertificateKeystore("PKCS12", toByteArray(Secret.fromString(value)), password);
+                byte[] keystoreBytes;
+
+                // JENKINS-41946: value can have 2 different formats
+                if (value.startsWith("{") && value.endsWith("}")) {
```

Comment:
this will not be sufficient as on latest cores all secrets, even ones from `Secret` will start and end with `{` and `}`

I think we need to ensure a consistent usage instead rather than trying to dynamically detect